### PR TITLE
Add JavaScript error report

### PR DESCRIPTION
**Pull request by ChristophWurst**
Fixes https://github.com/ChristophWurst/nextcloud_sentry/issues/20

Todo:
- [x] Find a way around the CSP restriction, currently our CSP does not allow access to any other domains.

cc @georgehrke

### CODE REVIEW

**Comment by rullzer** [comment]
Wouldn't it be easiest to just have a simple controller that just proxies the stuff to sentry? That way you get around the CSP ;)

**Comment by rullzer** [comment]
Else you can of course when this app is enabled just add a policy to the CSP

see https://github.com/nextcloud/files_texteditor/blob/master/appinfo/app.php#L14

**Comment by ChristophWurst** [comment]
> Wouldn't it be easiest to just have a simple controller that just proxies the stuff to sentry? That way you get around the CSP ;)

I think there's a lot of dark magic in the Sentry client to gather lots of information about the error and the browser for the report. I'd have to implement all that by myself.

**Comment by rullzer** [comment]
Ah the just extend the CSP when this app is enabled ;)

**Comment by ChristophWurst** [comment]
Will do, thanks for the tip!

**Comment by ChristophWurst** [comment]
CSP is fixed now. @rullzer @georgehrke @MorrisJobke give this a quick review please :)

File: lib/Controller/ScriptController.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		parent::__construct('sentry', $request);
+		$this->config = $config;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function dsn(): Response {
+		$pubDsn = $this->config->getSystemValue('sentry.public-dsn', null);
```
Comment (MorrisJobke) [inline lib/Controller/ScriptController.php]: This should be documented in the readme and maybe also add a link to https://docs.sentry.io/clients/javascript/usage/#preventing-abuse there.

Comment (ChristophWurst) [reply lib/Controller/ScriptController.php]: Good catch! I fixed it in the latest commit.

**Pull request merged**